### ISSUE TITLE
`unnecessary_cast` lint suggestion causes error

### ISSUE THREAD

**Issue by 64kramsystem** [rust-lang/rust-clippy#9380]
### Summary

In some cases, the `unnecessary_cast` lint suggestion causes error.

p.s. Rust rocks 😁

### Reproducer

I tried this code:

```rust
#[warn(clippy::unnecessary_cast)]

fn main() {
    let _x: i32 = -(1) as i32;
}
```

This happened:

```
warning: failed to automatically apply fixes suggested by rustc to crate `plrs`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see 
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error: expected one of `.`, `;`, `?`, `else`, or an operator, found `_i32`
 --> src/main.rs:4:23
  |
4 |     let _x: i32 = -(1)_i32;
  |                       ^^^^ expected one of `.`, `;`, `?`, `else`, or an operator

error: aborting due to previous error

Original diagnostics will follow.

warning: casting integer literal to `i32` is unnecessary
 --> src/main.rs:4:19
  |
4 |     let _x: i32 = -(1) as i32;
  |                   ^^^^^^^^^^^ help: try: `-(1)_i32`
  |
note: the lint level is defined here
 --> src/main.rs:1:8
  |
1 | #[warn(clippy::unnecessary_cast)]
  |        ^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast
```

### Version

```text
rustc 1.64.0-nightly (fe3342816 2022-08-01)
binary: rustc
commit-hash: fe3342816a282949f014caa05ea2e669ff9d3d3c
commit-date: 2022-08-01
host: x86_64-unknown-linux-gnu
release: 1.64.0-nightly
LLVM version: 14.0.6
```

### Additional Labels

@rustbot label + l-suggestion-causes-error
